### PR TITLE
chore: Inbox view enhancements

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

This PR includes
1. Fix the back button to redirect to the inbox view and separate the expanded layout that is used in the conversation screen.
2. Uses notification ID for the route instead of conversation ID.
3. Close the inbox list display and options menu if the inbox card context menu is open.
4. Removed the list end message.
5. Add snooze option in context menu if the route has notification id and the selected is not snoozed.

Fixes 
1. https://linear.app/chatwoot/issue/CW-3061/why-is-there-an-expanded-view-in-inbox
2. https://linear.app/chatwoot/issue/CW-3062/the-url-uses-conversation-id-it-should-be-the-notification-id
3. https://linear.app/chatwoot/issue/CW-3063/dropdown-doesnt-work-properly-overlapping-menus-can-be-seen
4. https://linear.app/chatwoot/issue/CW-3066/remove-the-end-of-list-message-it-is-not-required 
6. https://linear.app/chatwoot/issue/CW-3067/i-cant-snooze-on-right-click

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

**Loom video**
https://www.loom.com/share/90d665233dc44fe1835765d3f328d905?sid=84f0ad33-6b2d-4978-80f4-dcd7c75df1e8

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by muhsin-k — APPROVED** [review]

File: app/javascript/dashboard/routes/dashboard/inbox/components/InboxCard.vue [hunk 1]
```diff
@@ -131,6 +140,12 @@ export default {
         },
       ];
 
+      if (this.isNotSnoozed) {
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/inbox/components/InboxCard.vue:143]: Let us discuss this. I'm not sure about managing this in the view.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Please see the individual changes.

File: app/javascript/dashboard/routes/dashboard/inbox/InboxView.vue [hunk 2]
```diff
@@ -93,36 +100,23 @@ export default {
...
-        Boolean(this.conversationId) &&
-        Boolean(this.currentChat.id) &&
-        !this.isFetchingInitialData
-      );
+      return Boolean(this.notificationId) && !this.isFetchingInitialData;
     },
     totalNotifications() {
       return this.notifications?.length ?? 0;
     },
     activeNotificationIndex() {
```
[the remark this answers is not in the document]
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/inbox/InboxView.vue:114]: Done

File: app/javascript/dashboard/routes/dashboard/inbox/InboxView.vue [hunk 3]
```diff
@@ -82,9 +88,10 @@ export default {
       uiFlags: 'notifications/getUIFlags',
     }),
     activeNotification() {
-      return this.notifications.find(
-        n => n.primary_actor.id === Number(this.conversationId)
-      );
+      return this.notifications.find(n => n.id === Number(this.notificationId));
```
[the remark this answers is not in the document]
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/inbox/InboxView.vue:91]: Done

File: app/javascript/dashboard/routes/dashboard/inbox/InboxView.vue [hunk 4]
```diff
@@ -149,28 +145,42 @@ export default {
...
         .then(() => {
           bus.$emit(BUS_EVENTS.SCROLL_TO_MESSAGE);
         });
     },
     findConversation() {
-      const conversationId = Number(this.conversationId);
-      return this.allConversation.find(c => c.id === conversationId);
+      return this.allConversation.find(
+        c => c.id === Number(this.conversationId)
+      );
```
[the remark this answers is not in the document]
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/inbox/InboxView.vue:183]: Done

File: app/javascript/dashboard/routes/dashboard/inbox/components/InboxCard.vue (hunk 1 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/inbox/components/InboxCard.vue:143]: Sure

File: app/javascript/dashboard/routes/dashboard/inbox/InboxView.vue [hunk 5]
```diff
@@ -93,36 +100,23 @@ export default {
...
     isFetchingInitialData() {
       return this.uiFlags.isFetching && !this.notifications.length;
     },
     showInboxMessageView() {
-      return (
-        Boolean(this.conversationId) &&
-        Boolean(this.currentChat.id) &&
-        !this.isFetchingInitialData
-      );
+      return Boolean(this.notificationId) && !this.isFetchingInitialData;
```
[the remark this answers is not in the document]
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/inbox/InboxView.vue]: @pranavrajs Done https://github.com/chatwoot/chatwoot/pull/8898/commits/8de4911fb3bd7e8303b4a88b2a31ac9ef79ae205

File: app/javascript/dashboard/routes/dashboard/inbox/components/InboxCard.vue (hunk 1 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/inbox/components/InboxCard.vue:143]: @pranavrajs I have made some changes here, But in a sperate PR for easy review https://github.com/chatwoot/chatwoot/pull/8924

File: app/javascript/dashboard/routes/dashboard/inbox/InboxDetail.vue [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    notificationId() {
+      return Number(this.$route.params.notification_id);
+    },
+    activeNotification() {
+      return this.activeNotificationById(this.notificationId);
+    },
+    conversationId() {
+      return this.activeNotification?.primary_actor?.id;
+    },
+    isFetchingInitialData() {
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/inbox/InboxDetail.vue:55]: Unused property.

**Pull request closed without merging**